google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/facet/AppEngineStandardFacetEditor.java

```diff
@@ -167,50 +168,42 @@ private boolean isManagedLibrariesEnabled() {
-          .getLibraryByMavenDisplayName(addedLibrary.getName());
+      Optional<AppEngineStandardMavenLibrary> libraryOptional =
+          AppEngineStandardMavenLibrary.getLibraryByMavenDisplayName(addedLibrary.getName());
 
-      if (library == null) {
-        return;
-      }
+      libraryOptional.ifPresent(
+          library -> {
+            final DependencyScope scope = library.getScope();
 
-      final DependencyScope scope = library.getScope();
+            ApplicationManager.getApplication().runWriteAction(() -> {
+              ModifiableRootModel model =
+                  ModuleRootManager.getInstance(context.getModule()).getModifiableModel();
 
-      new WriteAction() {
-        @Override
-        protected void run(@NotNull Result result) throws Throwable {
-          ModuleRootManager manager = ModuleRootManager.getInstance(context.getModule());
-          ModifiableRootModel model = manager.getModifiableModel();
+              model.addLibraryEntry(addedLibrary);
 
-          model.addLibraryEntry(addedLibrary);
+              Arrays.stream(model.getOrderEntries())
+                  .filter(
+                      orderEntry -> orderEntry.getPresentableName().equals(addedLibrary.getName()))
+                  .forEach(orderEntry -> ((ExportableOrderEntry) orderEntry).setScope(scope));
 
-          for (OrderEntry orderEntry : model.getOrderEntries()) {
-            if (orderEntry.getPresentableName().equals(addedLibrary.getName())) {
-              ((ExportableOrderEntry) orderEntry).setScope(scope);
-            }
-          }
-          model.commit();
-        }
-      }.execute();
+              model.commit();
+            });
 
-      Artifact artifact = AppEngineSupportProvider
-          .findOrCreateWebArtifact((AppEngineStandardFacet) context.getFacet());
-      AppEngineStandardWebIntegration.getInstance()
-          .addLibraryToArtifact(addedLibrary, artifact, context.getProject());
+            Artifact artifact = AppEngineSupportProvider
+                .findOrCreateWebArtifact((AppEngineStandardFacet) context.getFacet());
+            AppEngineStandardWebIntegration.getInstance()
+                .addLibraryToArtifact(addedLibrary, artifact, context.getProject());
 
-      appEngineStandardLibraryPanel.toggleLibrary(
-          AppEngineStandardMavenLibrary.getLibraryByMavenDisplayName(addedLibrary.getName()),
-          true /* select */);
+            appEngineStandardLibraryPanel.toggleLibrary(library, true /* select */);
+          }
+      );
     }
 
     @Override
     public void afterLibraryRemoved(Library removedLibrary) {
-      AppEngineStandardMavenLibrary library = AppEngineStandardMavenLibrary
-          .getLibraryByMavenDisplayName(removedLibrary.getName());
-
-      if (library != null) {
-        appEngineStandardLibraryPanel.toggleLibrary(library, false /* select */);
-      }
+      AppEngineStandardMavenLibrary.getLibraryByMavenDisplayName(removedLibrary.getName())
```

Comment:
minor, but for consistency with the other method could you extract a variable here too?